pkg/installer/custominstallconfig_test.go

```diff
@@ -0,0 +1,328 @@
+	client.EXPECT().GetVMCapabilities(gomock.Any(), "Standard_D2s_v3", "centralus").
+		Return(map[string]string{
+			"vCPUsAvailable":               "4",
+			"MemoryGB":                     "16",
+			"PremiumIO":                    "True",
+			"HyperVGenerations":            "V1,V2",
+			"AcceleratedNetworkingEnabled": "True",
+			"CPUArchitectureType":          "x64",
+		}, nil).
+		AnyTimes()
+	client.EXPECT().GetMarketplaceImage(gomock.Any(), "centralus", "azureopenshift", "aro4", "aro_416", "416.00.20240517").
+		Return(compute.VirtualMachineImage{
+			VirtualMachineImageProperties: &compute.VirtualMachineImageProperties{
+				HyperVGeneration: compute.HyperVGenerationTypesV2,
+			},
+			Name:     to.StringPtr("aro_416"),
+			Location: to.StringPtr("centralus"),
+		}, nil).
+		AnyTimes()
+	client.EXPECT().GetGroup(gomock.Any(), "test-resource-group").
+		Return(&resources.Group{
+			ID:       to.StringPtr("test-resource-group"),
+			Location: to.StringPtr("centralus"),
+		}, nil)
+	client.EXPECT().ListResourceIDsByGroup(gomock.Any(), "test-resource-group").
+		Return([]string{}, nil)
+	client.EXPECT().GetHyperVGenerationVersion(gomock.Any(), "Standard_D2s_v3", "centralus", "").
+		Return("V2", nil)
+}
+
+func TestApplyInstallConfigCustomisations(t *testing.T) {
+	m := fakeManager()
+	inInstallConfig := makeInstallConfig()
+
+	mockCtrl := gomock.NewController(t)
+	defer mockCtrl.Finish()
+	mockClient := azuremock.NewMockAPI(mockCtrl)
+	inInstallConfig.Azure.UseMockClient(mockClient)
+	mockClientCalls(mockClient)
+
+	graph, err := m.applyInstallConfigCustomisations(inInstallConfig, makeImage())
+	if err != nil {
+		t.Fatal(err)
+	}
+
+	bootstrapAsset := graph.Get(&bootstrap.Bootstrap{}).(*bootstrap.Bootstrap)
+	bootstrapIgnition := string(bootstrapAsset.Files()[0].Data)
+
+	t.Log(bootstrapIgnition)
+}
+
+func TestApplyIgnitionConfigCustomisations(t *testing.T) {
+	m := fakeManager()
+	inInstallConfig := makeInstallConfig()
+
+	mockCtrl := gomock.NewController(t)
+	defer mockCtrl.Finish()
+	mockClient := azuremock.NewMockAPI(mockCtrl)
+	inInstallConfig.Azure.UseMockClient(mockClient)
+	mockClientCalls(mockClient)
+
+	graph, err := m.applyInstallConfigCustomisations(inInstallConfig, makeImage())
+	if err != nil {
+		t.Fatal(err)
+	}
+
+	graph, err = m.applyIgnitionConfigCustomisations(graph)
+	if err != nil {
+		t.Fatal(err)
+	}
+
+	bootstrapAsset := graph.Get(&bootstrap.Bootstrap{}).(*bootstrap.Bootstrap)
+	var temp map[string]any
+	err = json.Unmarshal(bootstrapAsset.Files()[0].Data, &temp)
+	if err != nil {
+		t.Fatal(err)
+	}
+
+	files := (temp["storage"].(map[string]any))["files"].([]any)
+	systemd := (temp["systemd"].(map[string]any))["units"].([]any)
+	storageFileList := map[string]bool{}
+	for _, file := range files {
+		storageFileList[file.(map[string]any)["path"].(string)] = true
+	}
+	systemdFileList := map[string]bool{}
+	for _, file := range systemd {
+		systemdFileList[file.(map[string]any)["name"].(string)] = true
+	}
```

Comment:
Does the fork do anything with the master/worker ignition stubs?
It might be worth putting this code in a little utility function.

Reply:
Not yet but might need it in the near future. Made it a function.